src/validation.cpp

```diff
@@ -4926,7 +4926,7 @@ bool CChainState::RollforwardBlock(const CBlockIndex* pindex, CCoinsViewCache& i
     std::vector<std::pair<CAddressUnspentKey, CAddressUnspentValue> > addressUnspentIndex;
     std::vector<std::pair<CSpentIndexKey, CSpentIndexValue> > spentIndex;
 
-    for (unsigned int i = 0; i < block.vtx.size(); i++) {
+    for (size_t i = 0; i < block.vtx.size(); i++) {
```

Comment:
why not auto?

Reply:
I quite don't agree about `auto` here. 
IMO 2 points to use auto: 
 - if type is complex such as `std::vector<std::pair<CAddressUnspendKey, CAddressUnspentValue>>::const_iterator` or something like that
 - if it is not easy to deduct type and specify it explitely

More over, changing type `size_t` to  `auto` here makes `i` initialized by `int` that is not quite expected. because it is initialized from `0`.

The code here: doesn't become simpler, doesn't become shorter, doesn't become clearer, moreover this definition becomes incorrect + comparision signed/unsigned variables in cycle + wrong size (4 bytes vs 8 bytes).

Please, return back explicit `size_t` here.